### PR TITLE
add trigger-now.jsp page

### PR DESCRIPTION
**Pull request by IsharaSandeepanie**

### CODE REVIEW

File: components/org.wso2.carbon.identity.remotefetch.core.ui/src/main/resources/web/remotefetch-core/trigger-now.jsp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    PollingActionListener pollingActionListener = new PollingActionListener(repositoryManager, configDeployer, 0,
+            fetchConfiguration.getRemoteFetchConfigurationId(), fetchConfiguration.getTenantId(), fetchConfiguration.getUserName());
+    
+    pollingActionListener.iteration();
+
+%>
+
+<script>
+    location.href = 'list-remotefetch-configs.jsp';
+</script>
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.remotefetch.core.ui/src/main/resources/web/remotefetch-core/trigger-now.jsp]: Add a new line

File: components/org.wso2.carbon.identity.remotefetch.core.ui/src/main/resources/org/wso2/carbon/identity/remotefetch/core/ui/i18n/Resources.properties [hunk 2]
```diff
@@ -15,19 +15,18 @@
...
+remotefetch.edit=Edit Remote Repository Deployment Configuration
+field.config.name=Deployment Name
+field.config.enabled=Enabled
+field.config.status=Status
+field.config.remoteFetchName=Repository Configuration Name
+field.config.repositoryType=Repository Type
+field.config.actionListenerType=Action Listener Type
+field.config.configarationDeployerType=Configuration Deployer Type
+field.config.userName=User Name
+field.config.action=Actions
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.remotefetch.core.ui/src/main/resources/org/wso2/carbon/identity/remotefetch/core/ui/i18n/Resources.properties]: Add a new line

File: components/org.wso2.carbon.identity.remotefetch.core.ui/src/main/resources/web/remotefetch-core/trigger-now.jsp (hunk 1 above)
Comment (IsharaSandeepanie) [reply components/org.wso2.carbon.identity.remotefetch.core.ui/src/main/resources/web/remotefetch-core/trigger-now.jsp]: [fixed commit](https://github.com/wso2-extensions/identity-fetch-remote/pull/12/commits/b011c6cd9f5715285a8608026f18878876b1f0b1)

File: components/org.wso2.carbon.identity.remotefetch.core.ui/src/main/resources/org/wso2/carbon/identity/remotefetch/core/ui/i18n/Resources.properties (hunk 2 above)
Comment (IsharaSandeepanie) [reply components/org.wso2.carbon.identity.remotefetch.core.ui/src/main/resources/org/wso2/carbon/identity/remotefetch/core/ui/i18n/Resources.properties]: [fixed commit](https://github.com/wso2-extensions/identity-fetch-remote/pull/12/commits/b011c6cd9f5715285a8608026f18878876b1f0b1)

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/implementations/configDeployers/VelocityTemplatedSPDeployer.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/implementations/configDeployers/VelocityTemplatedSPDeployer.java]: Update the licese header.

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/implementations/configDeployers/VelocityTemplatedSPDeployer.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class VelocityTemplatedSPDeployer extends ServiceProviderConfigDeployer {
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/implementations/configDeployers/VelocityTemplatedSPDeployer.java]: Add class comments

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/implementations/configDeployers/VelocityTemplatedSPDeployer.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    private int id;
+
+    public VelocityTemplatedSPDeployer(int tenantId, String userName, int id) {
+
+        super(tenantId, userName, id);
+        this.id = id;
+    }
+
+    @Override
+    public void deploy(ConfigurationFileStream configurationFileStream) throws RemoteFetchCoreException {
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/implementations/configDeployers/VelocityTemplatedSPDeployer.java]: Add method comments

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfiguration.java [hunk 6]
```diff
@@ -60,6 +63,22 @@ public int getTenantId() {
...
+    /**
+     * @param remoteFetchName
+     */
+    public void setRemoteFetchName(String remoteFetchName) {
+
+        this.remoteFetchName = remoteFetchName;
+    }
+
+    /**
+     * @return
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfiguration.java]: Add method comments

File: pom.xml [hunk 7]
```diff
@@ -103,6 +103,16 @@
                     </exclusion>
                 </exclusions>
             </dependency>
+            <dependency>
+                <groupId>com.cedarsoftware</groupId>
+                <artifactId>java-util</artifactId>
+                <version>1.8.0</version>
```
Comment (ashensw) [inline pom.xml]: Move the versions to parent pom.

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/implementations/configDeployers/VelocityTemplatedSPDeployer.java (hunk 3 above)
Comment (IsharaSandeepanie) [reply components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/implementations/configDeployers/VelocityTemplatedSPDeployer.java]: [Fixed](https://github.com/wso2-extensions/identity-fetch-remote/pull/12/commits/e33283f45ce3ac37a3c2ef622918b08c838285d4)

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfiguration.java (hunk 6 above)
Comment (IsharaSandeepanie) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfiguration.java]: [Fixed](https://github.com/wso2-extensions/identity-fetch-remote/pull/12/commits/e33283f45ce3ac37a3c2ef622918b08c838285d4)

File: pom.xml (hunk 7 above)
Comment (IsharaSandeepanie) [reply pom.xml]: [Fixed commit](https://github.com/wso2-extensions/identity-fetch-remote/pull/12/commits/e33283f45ce3ac37a3c2ef622918b08c838285d4)

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/implementations/configDeployers/VelocityTemplatedSPDeployer.java (hunk 4 above)
Comment (IsharaSandeepanie) [reply components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/implementations/configDeployers/VelocityTemplatedSPDeployer.java]: [Fixed commit](https://github.com/wso2-extensions/identity-fetch-remote/pull/12/commits/e33283f45ce3ac37a3c2ef622918b08c838285d4)

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/implementations/configDeployers/VelocityTemplatedSPDeployer.java (hunk 5 above)
Comment (IsharaSandeepanie) [reply components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/implementations/configDeployers/VelocityTemplatedSPDeployer.java]: [Fixed](https://github.com/wso2-extensions/identity-fetch-remote/pull/12/commits/e33283f45ce3ac37a3c2ef622918b08c838285d4)

**Pull request merged**